espressomd/espresso
# Unit testing tutorials and samples

Fixes #1377

Description of changes:
- **tests for tutorials and samples**
- **fixed broken samples and tutorial scripts**
- cleaned up tutorials and samples
- converted tutorial 8 to IPython


PR Checklist
------------
 - [x] Write tests (`/testsuite/scripts`)
 - [x] Document tests (`/doc/sphinx/testing.rst`)
 - [x] Integrate tests to CI (#1377):
   - update containers with missing Python packages:
     - nbconvert, matplotlib, scipy, mock (only Python2)
     - see error log at https://travis-ci.org/jngrad/espresso/builds/481848614
   - check containers have CMake version 3.7.0 or above (for test fixtures)
     - for example in Travis-CI job logs, system_info states CMake 3.9.2 when in reality it's 3.5.1
   - maybe schedule these tests differently (12 min total runtime)
